src/mscorlib/shared/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -90,5 +223,18 @@ public static bool TryGetArray<T>(ReadOnlyMemory<T> readOnlyMemory, out ArraySeg
             arraySegment = default;
             return false;
         }
+
+        /// <summary>
+        /// Creates an <see cref="IEnumerable{T}"/> view of the given <paramref name="memory" /> to allow
+        /// the <paramref name="memory" /> to be used in existing APIs that take an <see cref="IEnumerable{T}"/>.
+        /// </summary>
+        /// <typeparam name="T">The element type of the <paramref name="memory" />.</typeparam>
+        /// <param name="memory">The Memory to view as an <see cref="IEnumerable{T}"/></param>
+        /// <returns>An <see cref="IEnumerable{T}"/> view of the given <paramref name="memory" /></returns>
+        public static IEnumerable<T> ToEnumerable<T>(Memory<T> memory)
```

Comment:
Does this end up having the desired optimization?  I would hope so, as Environment.CurrentManagedThreadId is available to it, but the Environment it has access to is also internal, so we should verify that's not throwing off the compiler.

Reply:
@stephentoub `Environment.CurrentManagedThreadId` is not used here. Was this meant for something else?

Reply:
It's used by the compiler's implementation of iterators, and the compiler keys off its presence to determine whether to take the optimization that reuses the enumerable as the enumerator.

Reply:
(I expect it's fine though, as corelib already has other iterators in it, and they look to compile "correctly".)

Reply:
I checked the IL and confirmed that it is getting compiled correctly with the use of ```System.Environment::get_CurrentManagedThreadId()```